skyportal_fink_client/skyportal_fink_client.py

```diff
@@ -0,0 +1,150 @@
+# coding: utf-8
+import os
+import yaml
+from fink_client.consumer import AlertConsumer
+from astropy.time import Time
+
+import utils.skyportal_api as skyportal_api
+
+# open yaml config file
+with open(
+    os.path.abspath(os.path.join(os.path.dirname(__file__))) + "/../config.yaml", "r"
+) as stream:
+    try:
+        conf = yaml.safe_load(stream)
+    except yaml.YAMLError as exc:
+        print(exc)
+
+
+def fid_to_filter(fid: int):
+    """
+    Convert a fid to a filter name.
+    In the alert data from Fink, the fid corresponds to the 3 different filters used by the ZTF telescope.
+
+    Parameters
+    ----------
+    fid: int
+        id of a filter in an alert
+
+    Returns
+    ----------
+    filter: str
+        name of the filter
+    """
+    switcher = {1: "ztfg", 2: "ztfr", 3: "ztfi"}
+    return switcher.get(fid)
+
+
+def poll_alerts():
+    """
+    Connect to and poll alerts from fink servers to post them in skyportal using its API, using a config file containing
+    the necessary access credentials to both fink and skyportal, as well as a list of topics to subscribe to
+    (corresponding to a classification in skyportal).
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    ----------
+    None
+    """
+    myconfig = {
+        "username": conf["username"],
+        "bootstrap.servers": conf["servers"],
+        "group_id": conf["group_id"],
+    }
+
+    if conf["password"] is not None:
+        myconfig["password"] = conf["password"]
+
+    # extract all topics from conf['mytopics'] and create a list of topics names
+    topics = list(conf["mytopics"].keys())
+    print(f"Fink topics you subscribed to: {topics}")
+
+    taxonomy_ids = {}
+    for topic in topics:
+        status, taxonomy_id = skyportal_api.get_taxonomy_id_including_classification(
+            conf["mytopics"][topic]["classification"],
+            conf["skyportal_url"],
+            conf["skyportal_token"],
+        )
+        if status != 401:
+            if taxonomy_id is not None:
+                taxonomy_ids[topic] = taxonomy_id
+            else:
+                return print(
+                    "Classification not found in taxonomy:",
+                    conf["mytopics"][topic]["classification"],
+                )
+        else:
+            return print("Skyportal token not valid")
+
+    fink_id, stream_id, filter_id = skyportal_api.init_skyportal(
+        conf["skyportal_url"], conf["skyportal_token"]
+    )
+
+    maxtimeout = 5
```

Comment:
Make this as an argument of the function with a default of 5 seconds

Reply:
Agreed. Will change that now.